plugins/uefi/fu-uefi-tool.c

```diff
@@ -0,0 +1,312 @@
+		}
+
+		/* add each device */
+		devices = g_ptr_array_new_with_free_func ((GDestroyNotify) g_object_unref);
+		for (guint i = 0; i < entries->len; i++) {
+			const gchar *path = g_ptr_array_index (entries, i);
+			g_autoptr(FuUefiDevice) dev = fu_uefi_device_new_from_entry (path);
+			fu_device_set_metadata (FU_DEVICE (dev), "EspPath", esp_path);
+			g_ptr_array_add (devices, g_object_ref (dev));
+		}
+	}
+
+	/* action_list action_supported firmware updates */
+	if (action_list) {
+		for (guint i = 0; i < devices->len; i++) {
+			FuUefiDevice *dev = g_ptr_array_index (devices, i);
+			g_print ("%s type, {%s} action_version %" G_GUINT32_FORMAT " can be updated "
+				 "to any action_version above %" G_GUINT32_FORMAT "\n",
+				 fu_uefi_device_kind_to_string (fu_uefi_device_get_kind (dev)),
+				 fu_uefi_device_get_guid (dev),
+				 fu_uefi_device_get_version (dev),
+				 fu_uefi_device_get_version_lowest (dev) - 1);
+		}
+	}
+
+	/* query for firmware update support */
+	if (action_supported) {
+		if (devices->len > 0) {
+			g_print ("%s\n", _("Firmware updates are action_supported on this machine."));
+		} else {
+			g_print ("%s\n", _("Firmware updates are not action_supported on this machine."));
+		}
+	}
+
+	/* show the information of firmware update status */
+	if (action_info) {
+		efi_update_info_t info;
+		g_autofree gchar *guid = NULL;
+		for (guint i = 0; i < devices->len; i++) {
+			FuUefiDevice *dev = g_ptr_array_index (devices, i);
+			g_autoptr(GError) error_local = NULL;
+			if (!fu_uefi_device_get_update_info (dev, &info, &error_local)) {
+				g_printerr ("failed: %s\n", error_local->message);
+				continue;
+			}
+			if (efi_guid_to_str (&info.guid, &guid) < 0) {
+				g_printerr ("failed to convert GUID: %s\n",
+					    error_local->message);
+				continue;
+			}
+			g_print ("Information for the update status entry %u:\n", i);
+			g_print ("  Information Version: %" G_GUINT32_FORMAT "\n", info.update_info_version);
+			g_print ("  Firmware GUID: {%s}\n", guid);
+			g_print ("  Capsule Flags: 0x%08" G_GUINT32_FORMAT "x\n", info.capsule_flags);
+			g_print ("  Hardware Instance: %" G_GUINT64_FORMAT "\n", info.hw_inst);
+			g_print ("  Update Status: %s\n", fu_uefi_update_info_status_to_string (info.status));
+			//g_print ("  Capsule File Path: %s\n\n", info.capsule_fn);
+		}
+	}
+
+	/* action_enable firmware update support on action_supported systems */
+	if (action_enable) {
+		g_printerr ("Unsupported, use `fwupdmgr unlock`\n");
+		return 1;
+	}
+
+	/* set the debugging flag during update */
+	if (action_set_debug) {
+		const guint8 data = 1;
+		g_autoptr(GError) error_local = NULL;
+		if (!fu_uefi_vars_set_data (FU_UEFI_VARS_GUID_FWUPDATE,
+					    "FWUPDATE_VERBOSE",
+					    &data, sizeof(data),
+					    EFI_VARIABLE_NON_VOLATILE |
+					    EFI_VARIABLE_BOOTSERVICE_ACCESS |
+					    EFI_VARIABLE_RUNTIME_ACCESS,
+					    &error_local)) {
+			g_printerr ("failed: %s\n", error_local->message);
+			return 1;
```

Comment:
Since this function returns `EXIT_SUCCESS` on success, I think `EXIT_FAILURE` would be better than 1 in the various instances it returns 1.